homeassistant/components/fritzbox/climate.py

```diff
@@ -82,53 +86,53 @@ class FritzboxThermostat(FritzBoxEntity, ClimateEntity):
     """The thermostat class for FRITZ!SmartHome thermostates."""
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Return the list of supported features."""
         return SUPPORT_FLAGS
 
     @property
-    def available(self):
+    def available(self) -> bool:
         """Return if thermostat is available."""
-        return self.device.present
+        return self.device.present  # type: ignore [no-any-return]
 
     @property
-    def temperature_unit(self):
+    def temperature_unit(self) -> str:
         """Return the unit of measurement that is used."""
         return TEMP_CELSIUS
 
     @property
-    def precision(self):
+    def precision(self) -> float:
         """Return precision 0.5."""
         return PRECISION_HALVES
 
     @property
-    def current_temperature(self):
+    def current_temperature(self) -> float:
         """Return the current temperature."""
-        return self.device.actual_temperature
+        return self.device.actual_temperature  # type: ignore [no-any-return]
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> float:
         """Return the temperature we try to reach."""
         if self.device.target_temperature == ON_API_TEMPERATURE:
             return ON_REPORT_SET_TEMPERATURE
         if self.device.target_temperature == OFF_API_TEMPERATURE:
             return OFF_REPORT_SET_TEMPERATURE
-        return self.device.target_temperature
+        return self.device.target_temperature  # type: ignore [no-any-return]
 
-    async def async_set_temperature(self, **kwargs):
+    async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
-        if ATTR_HVAC_MODE in kwargs:
-            hvac_mode = kwargs.get(ATTR_HVAC_MODE)
+        if kwargs.get(ATTR_HVAC_MODE) is not None:
```

Comment:
Just create a proper typed argument with default value. If you can trace all calls to `async_set_temperature` you can eliminate `kwargs` completely.

Reply:
I like this idea, but with this get `error: Signature of "async_set_temperature" incompatible with supertype "ClimateEntity"` :confused: